pallets/rbac/src/functions.rs

```diff
@@ -15,8 +15,8 @@ impl<T: Config> RoleBasedAccessControl<T::AccountId> for Pallet<T>{
     /// ### Parameters:
     /// - `pallet_id`: The unique pallet identifier.
     /// - `scope_id`: The newly generated scope identifier.
-    fn create_scope(pallet_id: u64, scope_id: ScopeId)-> DispatchResult{
-        let pallet_id: u64 = pallet_id.try_into().unwrap();
+    fn create_scope(pallet_name: Vec<u8>, scope_id: ScopeId)-> DispatchResult{
```

Comment:
I think it would be good to handle the pallet_name/pallet_id parameter as an enum, that can be either the pallet_name or the pallet_id(the hashed pallet name), this enum could have to functions:
- to_id_enum() that returns the enum that contains a pallet_id
- to_id() that returns the actual pallet_id
This would enable:
- the encapsulation of all the logic related to this parameter in a type
- the hashing of the name only once in cases where a function that receives this parameter calls multiple other functions that also receives this parameter

What do you think?

Reply:
Interesting solution, I developed a similar solution for role name or role id, but I discarded it because it adds some development complexity, I'll try that out again

Reply:
How does this implementation look? I'm going to test this enum out
```
#[derive(Encode, Decode, Clone, Eq, PartialEq,)]
pub enum IdOrVec{
    Id([u8;32]),
    Vec(Vec<u8>)
}

impl IdOrVec{
    pub fn to_id_enum(&self)->Self{
        match self{
            Self::Id(_) => self.clone(),
            Self::Vec(_) => Self::Id(Self::to_id(&self))
        }
    }

    pub fn to_id(&self)->[u8;32]{
        match self{
            Self::Id(id) => id.clone(),
            Self::Vec(v) => v.clone().using_encoded(blake2_256)
        }
    }
}
```

Reply:
It looks good!